src/ucp/core/ucp_am.c

```diff
@@ -993,6 +960,12 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_am_send_nbx,
 
     UCP_WORKER_THREAD_CS_ENTER_CONDITIONAL(worker);
 
+    status = ucp_am_send_nbx_check_header_length(worker, header_length);
+    if (status != UCS_OK) {
+        ret = UCS_STATUS_PTR(status);
+        goto out;
+    }
```

Comment:
```suggestion
    if (ENABLE_PARAMS_CHECK && (header_length > worker->max_am_header)) {
        ucs_error("active message header length (%zi) is greater than maximum "
                  "allowed header size (%zi)",
                  header_length, worker->max_am_header);
        ret = UCS_STATUS_PTR(UCS_ERR_INVALID_PARAM);
        goto out;
    }
```

@brminich the variant with additional method looks prettier for me.
And there is another check in this method, so it looks also more unified.